### PR TITLE
tbprofiler v6.4.1

### PR DESCRIPTION
**Pull request by sage-wright**
Update TBProfiler to v6.4.1

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [ ] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by kapsakcj** [comment]
Simple change from previous dockerfile:
```
$ diff tbprofiler/6.4.0/Dockerfile tbprofiler/6.4.1/Dockerfile 
6c6
< ARG TBPROFILER_VER="6.4.0"
---
> ARG TBPROFILER_VER="6.4.1"
```

Looks like the test passed:
```
#12 [test 3/3] RUN wget -q ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR166/009/ERR1664619/ERR1664619_1.fastq.gz &&     wget -q ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR166/009/ERR1664619/ERR1664619_2.fastq.gz &&     tb-profiler profile -1 ERR1664619_1.fastq.gz -2 ERR1664619_2.fastq.gz -t 2 -p ERR1664619 --txt
#12 16.24 [19:00:09] INFO     Using ref file:                                    db.py:614
#12 16.24                     /opt/conda/share/tbprofiler//tbdb.fasta                     
#12 16.24            INFO     Using gff file:                                    db.py:614
#12 16.24                     /opt/conda/share/tbprofiler//tbdb.gff                       
#12 16.24            INFO     Using bed file:                                    db.py:614
#12 16.24                     /opt/conda/share/tbprofiler//tbdb.bed                       
#12 16.24            INFO     Using json_db file:                                db.py:614
#12 16.24                     /opt/conda/share/tbprofiler//tbdb.dr.json                   
#12 16.36            INFO     Using variables file:                              db.py:614
#12 16.36                     /opt/conda/share/tbprofiler//tbdb.variables.json            
#12 16.37            INFO     Using spoligotype_spacers file:                    db.py:614
#12 16.37                     /opt/conda/share/tbprofiler//tbdb.spoligotype_spac          
#12 16.37                     ers.txt                                                     
#12 16.37            INFO     Using spoligotype_annotations file:                db.py:614
#12 16.37                     /opt/conda/share/tbprofiler//tbdb.spoligotype_list          
#12 16.37                     .csv                                                        
#12 16.37            INFO     Using bedmask file:                                db.py:614
#12 16.37                     /opt/conda/share/tbprofiler//tbdb.mask.bed                  
#12 16.37            INFO     Using barcode file:                                db.py:614
#12 16.37                     /opt/conda/share/tbprofiler//tbdb.barcode.bed               
#12 16.37            INFO     Using rules file:                                  db.py:614
#12 16.37                     /opt/conda/share/tbprofiler//tbdb.rules.txt                 
#12 16.37            INFO     Trimming reads                                   fastq.py:39
#12 19.18 [19:00:12] INFO     Mapping to reference genome                      fastq.py:53
#12 65.12 
Calling variants:   0%|          | 0/73 [00:00<?, ?it/s]
Calling variants:   1%|▏         | 1/73 [00:00<00:31,  2.25it/s]
Calling variants:   3%|▎         | 2/73 [00:00<00:17,  4.03it/s]
Calling variants:   5%|▌         | 4/73 [00:00<00:09,  7.66it/s]
Calling variants:   8%|▊         | 6/73 [00:00<00:06,  9.78it/s]
Calling variants:  11%|█         | 8/73 [00:00<00:05, 11.42it/s]
Calling variants:  14%|█▎        | 10/73 [00:01<00:04, 13.17it/s]
Calling variants:  16%|█▋        | 12/73 [00:01<00:04, 13.16it/s]
Calling variants:  19%|█▉        | 14/73 [00:01<00:04, 13.46it/s]
Calling variants:  22%|██▏       | 16/73 [00:01<00:04, 13.98it/s]
Calling variants:  25%|██▍       | 18/73 [00:01<00:03, 14.63it/s]
Calling variants:  27%|██▋       | 20/73 [00:01<00:03, 14.12it/s]
Calling variants:  30%|███       | 22/73 [00:01<00:03, 15.11it/s]
Calling variants:  33%|███▎      | 24/73 [00:01<00:03, 15.46it/s]
Calling variants:  36%|███▌      | 26/73 [00:02<00:03, 14.38it/s]
Calling variants:  38%|███▊      | 28/73 [00:02<00:03, 14.62it/s]
Calling variants:  41%|████      | 30/73 [00:02<00:02, 15.14it/s]
Calling variants:  44%|████▍     | 32/73 [00:02<00:02, 15.16it/s]
Calling variants:  47%|████▋     | 34/73 [00:02<00:02, 14.50it/s]
Calling variants:  49%|████▉     | 36/73 [00:02<00:02, 14.65it/s]
Calling variants:  52%|█████▏    | 38/73 [00:02<00:02, 14.81it/s]
Calling variants:  55%|█████▍    | 40/73 [00:03<00:02, 15.63it/s]
Calling variants:  58%|█████▊    | 42/73 [00:03<00:02, 15.45it/s]
Calling variants:  60%|██████    | 44/73 [00:03<00:01, 15.76it/s]
Calling variants:  63%|██████▎   | 46/73 [00:03<00:01, 15.58it/s]
Calling variants:  66%|██████▌   | 48/73 [00:03<00:01, 16.15it/s]
Calling variants:  68%|██████▊   | 50/73 [00:03<00:01, 16.18it/s]
Calling variants:  71%|███████   | 52/73 [00:03<00:01, 15.88it/s]
Calling variants:  74%|███████▍  | 54/73 [00:03<00:01, 15.97it/s]
Calling variants:  77%|███████▋  | 56/73 [00:04<00:01, 15.74it/s]
Calling variants:  79%|███████▉  | 58/73 [00:04<00:00, 15.90it/s]
Calling variants:  82%|████████▏ | 60/73 [00:04<00:00, 16.05it/s]
Calling variants:  85%|████████▍ | 62/73 [00:04<00:00, 15.03it/s]
Calling variants:  88%|████████▊ | 64/73 [00:04<00:00, 15.42it/s]
Calling variants:  90%|█████████ | 66/73 [00:04<00:00, 14.32it/s]
Calling variants:  93%|█████████▎| 68/73 [00:04<00:00, 14.56it/s]
Calling variants:  96%|█████████▌| 70/73 [00:05<00:00, 13.51it/s]
Calling variants:  99%|█████████▊| 72/73 [00:05<00:00, 13.94it/s]
Calling variants: 100%|██████████| 73/73 [00:05<00:00, 14.02it/s]
#12 70.33 
Indexing variants:   0%|          | 0/73 [00:00<?, ?it/s]
Indexing variants:  18%|█▊        | 13/73 [00:00<00:00, 127.99it/s]
Indexing variants:  40%|███▉      | 29/73 [00:00<00:00, 137.72it/s]
Indexing variants:  89%|████████▉ | 65/73 [00:00<00:00, 234.26it/s]
Indexing variants: 100%|██████████| 73/73 [00:00<00:00, 205.53it/s]
#12 70.86 [19:01:04] INFO     Running delly                                      bam.py:84
#12 74.86 [19:01:08] INFO     Running snpEff                                    vcf.py:126
#12 76.94 [19:01:10] INFO     Loading snpEff annotations                        vcf.py:164
#12 77.74 [19:01:11] INFO     Getting genotypes for positions in bed file       bam.py:291
#12 80.00 [19:01:13] INFO     Calculating bamstats                              bam.py:351
#12 82.58 [19:01:16] INFO     Getting qc metrics for regions                    bam.py:388
#12 82.58            INFO     Calculating depth in regions                       bam.py:54
#12 85.77 [19:01:19] INFO     Calculating median depth using samtools           bam.py:221
#12 87.96 [19:01:21] INFO     Getting missing genomic positions                 bam.py:381
#12 88.58 [19:01:22] INFO     Writing outputs                                  output.py:8
#12 88.58            INFO     Writing json file:                              output.py:15
#12 88.58                     /tests/results/ERR1664619.results.json                      
#12 88.59            INFO     Writing text file:                              output.py:19
#12 88.59                     /tests/results/ERR1664619.results.txt                       
#12 88.67            INFO     Profiling finished sucessfully!              tb-profiler:221
#12 DONE 88.9s
```

File: tbprofiler/6.4.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# Install dependencies via apt-get; cleanup apt garbage
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    wget \
+    ca-certificates \
+    procps && \
+    apt-get autoclean && rm -rf /var/lib/apt/lists/*
+
+# install tb-profiler via bioconda; install into 'base' conda env
+RUN micromamba install --yes --name base --channel conda-forge --channel bioconda  \
+    tb-profiler=${TBPROFILER_VER} delly=1.2.6 && \
```
Comment (kapsakcj) [inline tbprofiler/6.4.1/Dockerfile:37]: I think we can remove `delly=1.2.6` now that the pathogen-profiler recipe pins delly 1.2.6: 

https://github.com/bioconda/bioconda-recipes/blob/master/recipes/pathogen-profiler/meta.yaml#L34

Doesn't really make a difference unless Jody plans on upgrading pathogen-profiler to use a higher delly version in the future (unsure how likely that is though...)

up to you if you want to change or not. I don't think it makes a difference right now

Comment (sage-wright) [reply tbprofiler/6.4.1/Dockerfile:37]: nah, i think we're probably fine vfor now

Comment (sage-wright) [reply tbprofiler/6.4.1/Dockerfile:37]: but i'll remove it so that future people don't automatically assume to use this version

**Comment by kapsakcj** [comment]
I re-launched the deploy workflow. I think it's failing in the test layer when downloading the test FASTQs. Likely just the download from ENA causing the error

**Pull request merged**